core/node/api_server/src/web3/namespaces/en.rs

```diff
@@ -14,16 +15,60 @@ use crate::web3::{backend_jsonrpsee::MethodTracer, state::RpcState};
 #[derive(Debug)]
 pub(crate) struct EnNamespace {
     state: RpcState,
+    /// First batch to commit to L1 simulated by the main node.
+    /// This is temporary and used only for testing L1 batch signing by consensus attesters.
+    first_batch_to_commit: L1BatchNumber,
+}
+
+fn to_l1_batch_number(n: attester::BatchNumber) -> anyhow::Result<L1BatchNumber> {
+    Ok(L1BatchNumber(
+        n.0.try_into().context("L1BatchNumber overflow")?,
+    ))
 }
 
 impl EnNamespace {
-    pub fn new(state: RpcState) -> Self {
-        Self { state }
+    pub async fn new(state: RpcState) -> anyhow::Result<Self> {
+        let first_batch_to_commit = async {
+            let mut conn = state.acquire_connection().await.context("connection()")?;
+            // Try to continue from where we left.
+            if let Some(last) = conn
+                .consensus_dal()
+                .get_last_batch_certificate_number()
+                .await
+                .context("get_last_batch_certificate_number()")?
+            {
+                return to_l1_batch_number(last + 1);
+            }
+            // Otherwise start with the next sealed L1 batch.
+            if let Some(sealed) = conn
+                .blocks_dal()
+                .get_sealed_l1_batch_number()
+                .await
+                .context("get_sealed_l1_batch_number()")?
+            {
+                return Ok(sealed + 1);
```

Comment:
Why is it the last sealed + 1, why not the last sealed itself? How do we sign the fist batch if it already exists when `EnNamespace::new` is called? Say we have batch 0 in the table but no certificate for it, why should we start signing from 1?

Reply:
it doesn't matter whether it is +0,+1,+5,... this is just a starting point and it can be arbitrarily delayed.

Reply:
Okay, since we're not aiming to fill in the history, it doesn't matter. Might want to leave a comment to explain that it's an arbitrary choice.

Reply:
the +1 implementation makes the behavior with empty and nonempty set of sealed batches consistent.